code/game/machinery/shieldgen.dm

```diff
@@ -305,142 +305,115 @@
 	anchored = FALSE
 	density = TRUE
 	req_access = list(ACCESS_TELEPORTER)
-	var/active = FALSE
-	var/power = FALSE
-	var/state = FALSE
-	var/steps = 0
+	var/activated = FALSE
 	var/locked = TRUE
-	var/storedpower = 0
+	var/list/active_shields
+	var/stored_power = 0
 	flags = CONDUCT
 	use_power = NO_POWER_USE
 
+/obj/machinery/shieldwallgen/Initialize(mapload)
+	. = ..()
+	active_shields = list() // Doing it here since you can't cast numeral defines to strings using "[NORTH]" in the definition
+	for(var/direction in GLOB.cardinal)
+		active_shields["[direction]"] = list()
+
 /obj/machinery/shieldwallgen/update_icon_state()
-	icon_state = "Shield_Gen[active ? " +a" : ""]"
+	icon_state = "Shield_Gen[activated ? " +a" : ""]"
 
-/obj/machinery/shieldwallgen/proc/power()
-	if(!anchored)
-		power = FALSE
-		return
+/obj/machinery/shieldwallgen/proc/try_charge_shields_power()
 	var/turf/T = loc
 
 	var/obj/structure/cable/C = T.get_cable_node()
```

Comment:
can these fit in lockers and stuff? could cause issues if their loc isnt a turf.

Reply:
They should never be located in a locker or such. If this happens then something got fucked up majorly and it should cause a runtime